### PR TITLE
905 - List Active Offences endpoint

### PR DESCRIPTION
**Pull request by brownbearsoftware**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/community/Conviction.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  val offences: List<Offence>?
+)
+
+data class Offence(
+  val offenceId: String,
+  val mainOffence: Boolean,
+  val detail: OffenceDetail,
+  val offenceDate: LocalDateTime,
+  val offenceCount: Long,
+  val tics: Long,
```
Comment (edavey) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/community/Conviction.kt:25]: I think we'll need to mark `tics` as optional as I don't see it in responses I've obtained...

**Comment by brownbearsoftware** [comment]
[Select index offence (if there is > 1 active conviction)](https://trello.com/c/GE0uChM6/905-select-index-offence-if-there-is-1-active-conviction)

File: src/main/resources/static/api.yml [hunk 2]
```diff
@@ -3225,3 +3256,22 @@ components:
...
         - Completed
         - Incomplete
+    ActiveOffence:
+      type: object
+      properties:
+        deliusEventNumber:
+          type: string
+          example: "7"
+        offenceDescription:
+          type: string
```
Comment (edavey) [inline src/main/resources/static/api.yml:3266]: perhaps an example value would be helpful here? e.g.

`"Wounding or inflicting grievous bodily harm (inflicting bodily injury with or without weapon) (S20) - 00801"`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/ConvictionTransformer.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.ActiveOffence
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.community.Conviction
+
+@Component
+class ConvictionTransformer {
+  fun transformToApi(conviction: Conviction) = conviction.offences?.map {
+    ActiveOffence(
+      deliusEventNumber = conviction.index,
+      offenceDescription = "${it.detail.mainCategoryDescription} - ${it.detail.subCategoryDescription}",
```
Comment (edavey) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/ConvictionTransformer.kt:12]: 👍 this makes sense as sometimes (though not usually) the sub category description provides additional detail which may be important. Perhaps a "nice to have" would be a function which only includes the sub-category description if it is different from the main-category description...

**Review by edavey — APPROVED** [review]
👍 I've left a couple of minor comments.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/community/Conviction.kt (hunk 1 above)
Comment (brownbearsoftware) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/community/Conviction.kt:25]: Ah cheers for this!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/ConvictionTransformer.kt (hunk 3 above)
Comment (brownbearsoftware) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/ConvictionTransformer.kt]: Good shout - have added a check to see if they're the same

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/transformer/ConvictionTransformerTest.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      )
+      .produce()
+
+    assertThat(convictionTransformer.transformToApi(conviction)).containsExactlyInAnyOrder(
+      ActiveOffence(
+        deliusEventNumber = "5",
+        offenceDescription = "A Description",
+        offenceId = "1",
+        convictionId = 12345
+      )
```
Comment (edavey) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/transformer/ConvictionTransformerTest.kt:72]: nice, thanks for implementing!

**Review by edavey — COMMENTED** [review]
🚀

**Review by edavey — APPROVED** [review]

**Pull request merged**